lib/rubocop/rspec/expect_offense.rb

```diff
@@ -77,26 +93,31 @@ def self.parse(annotated_source)
           source      = []
           annotations = []
 
-          annotated_source.each_line do |source_line|
+          source_area, _seperator, correction =
+            annotated_source.partition(CORRECTION_SEPARATOR)
+          source_area.each_line do |source_line|
             if source_line =~ ANNOTATION_PATTERN
               annotations << [source.size, source_line]
             else
               source << source_line
             end
           end
 
-          new(source, annotations)
+          new(source, annotations, correction)
         end
 
+        attr_reader :correction
+
         # @param lines [Array<String>]
         # @param annotations [Array<(Integer, String)>]
         #   each entry is the annotated line number and the annotation text
         #
         # @note annotations are sorted so that reconstructing the annotation
         #   text via {#to_s} is deterministic
-        def initialize(lines, annotations)
+        def initialize(lines, annotations, correction = '')
           @lines       = lines.freeze
           @annotations = annotations.sort.freeze
+          @correction  = correction.empty? ? nil : correction.freeze
```

Comment:
I think we can just let the empty string communicate that there is no correction available. No need for a `nil`.